File: app/src/main/java/ru/deltadelete/lab10/entities/Country.java
package ru.deltadelete.lab10.entities;

import androidx.room.ColumnInfo;
import androidx.room.Entity;
import androidx.room.PrimaryKey;

@Entity(tableName = "countries")
public class Country {
    @PrimaryKey(autoGenerate = true)
    @ColumnInfo(name = "country_id")
    public int id;

    @ColumnInfo(name = "country_name")
    public String name;

    @ColumnInfo(name = "country_code")
    public String code;
}

File: app/src/main/java/ru/deltadelete/lab10/database/entities/Country.kt
package ru.deltadelete.lab10.database.entities

import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.PrimaryKey

@Entity(tableName = "countries")
data class Country (
    @PrimaryKey(autoGenerate = true)
    @ColumnInfo(name = "country_id")
    val id: Int = 0,

    @ColumnInfo(name = "country_name")
    val name: String,

    @ColumnInfo(name = "country_code")
    val code: String,
)


public val Country.flagUrl: String
    get() = "https://flagcdn.com/h120/${code}.png"